contracts/VaultV3.vy

```diff
@@ -606,24 +578,33 @@ def _process_report(strategy: address) -> (uint256, uint256):
     else:
         loss = current_debt - total_assets
 
+    # We compute unlocked_profit
+    profit_end_date: uint256 = self.profit_end_date
+    unlocked_profit: uint256 = 0 
+    if self.profit_distribution_rate_ != 0:
+        if block.timestamp > profit_end_date:
+            unlocked_profit = (profit_end_date - self.profit_last_update) * self.profit_distribution_rate_ / MAX_BPS
+            self.profit_distribution_rate_ = 0
+        else:
+            unlocked_profit = (block.timestamp - self.profit_last_update) * self.profit_distribution_rate_ / MAX_BPS
+
     remaining_time: uint256 = 0
-    if block.timestamp >= self.profit_end_date:
-        remaining_time = 0
-    else:
-        remaining_time = self.profit_end_date - block.timestamp
-    pending_profit: uint256 = self.profit_distribution_rate * remaining_time / MAX_BPS
+    if block.timestamp < profit_end_date:
+        remaining_time = profit_end_date - block.timestamp
+    pending_profit: uint256 = self.profit_distribution_rate_ * remaining_time / MAX_BPS
 
     if loss > 0:
         self.strategies[strategy].total_loss += loss
         self.strategies[strategy].current_debt -= loss
 
         if loss >= pending_profit:
             # If loss is too big for the profit buffer, we set distribution rate to zero
-            self.total_debt -= loss - pending_profit
-            self.profit_distribution_rate = 0
-            self.profit_end_date = 0
+            self.total_debt_ = self.total_debt_ - (loss - pending_profit) + unlocked_profit
```

Comment:
this is minor but I think it's generally a good idea to do all the additions first, then the substractions (`self.total_debt_ + unlocked_profit - (loss - pending_profit)`)

Reply:
done